src/views/index.pug

```diff
@@ -22,4 +22,49 @@ preview
         label(for='comments') Comments?
         textarea(id='comments' name='comments').form-control
       button(type='submit').btn.btn-primary Submit
+script(type='text/javascript')
+  | var tickingElements = [];
+  | function hookElement(id) {
+  |   var element = document.getElementById(id);
+  |   var elementObject = {
+  |     id: id,
+  |     element: element,
+  |     hasValue: false,
+  |   };
+  |   tickingElements.push(elementObject);
+  |
+  |   element.addEventListener('input', function () {
+  |     ensureListener((elementObject.hasValue = !!element.value));
+  |   });
+  | }
+  |
+  | var listening = false;
+  | function beforeUnload(event) {
+  |   event.preventDefault();
+  |   return (event.returnValue =
+  |     "You haven't sent your eval yet! Are you sure you want to leave?");
+  | }
+  |
+  | function ensureListener(definitelyValues) {
+  |   if (definitelyValues) {
+  |     if (!listening) {
+  |       window.addEventListener('beforeunload', beforeUnload, { capture: true });
+  |       listening = true;
+  |     }
+  |   } else if (
+  |     !tickingElements.some(function (element) {
+  |       return element.hasValue;
+  |     })
+  |   ) {
+  |     window.removeEventListener('beforeunload', beforeUnload, { capture: true });
+  |     listening = false;
+  |   }
+  | }
+  |
+  | hookElement('name');
+  | hookElement('position');
+  | hookElement('likes');
+  | hookElement('dislikes');
+  | hookElement('comments');
```

Comment:
Hmm, what if rather than listing all of them, could you just iterate on the textareas? I don't think name and position are all that important enough to warrant the prompt.

Reply:
Iterating over every textarea would be inefficient, DOM queries are slow and I believe internally element `id`s live in a map so they're not _as_ slow.

Reply:
This only runs once and the page is smol, so it probably wouldn't have much of an impact. That being said, sounds good to me.